revise YAML scripts and create YAML files for all subroutines in wetdep.F90

tangshuaiqi opened the pull request:
I created YAML files for all subroutines in wetdep.F90. Also, I revised the scripts to add more options that may be used in some subroutines:

1. add the code template if there should be multiple sets of YAML files to cover if/else conditions
2. sometimes the subroutine is called multiple times but we only need one set of YAML file. Add the option to choose which call is outputted.
3. add the code template to find the best lat/lon/level/timestep and the number of calls (n_calls) that gives good input/output variables.

These changes are in the `mam4_refactor_scripts/beg.inp` and `mam4_refactor_scripts/end.inp` files

Commit:
create YAML file for clddiag subroutine in wetdep.F90

Commit:
change yaml input variables to 1d vertical vars

Commit:
add yaml files for all subroutines in wetdep.F90

Commit:
revise the YAML beg and end scripts to allow more options such as selecting one call if the subroutine is called multiple times

singhbalwinder reviewed (approved):
Looks great! Thanks @tangshuaiqi for extending the capabilities!

For the "dependent" routines, we can simply use `if (yaml%flag_print)` since this flag will be set `true` by the parent subroutine.

tangshuaiqi commented:
> Looks great! Thanks @tangshuaiqi for extending the capabilities!
> 
> For the "dependent" routines, we can simply use `if (yaml%flag_print)` since this flag will be set `true` by the parent subroutine.

If I understand this correctly, I only need to pass the variable `yaml` from the parent subroutine to the inner subroutines and use 
`if (yaml%flag_print)` for the inner subroutines. Is that right? I will try it in later ones.

singhbalwinder commented:
Yes, that is correct. I think it makes it a lot simpler.

tangshuaiqi commented:
@singhbalwinder another question: when you do long-term test on the PR, did you run test with the current one (that not call the YAML subroutines) or add the YAML subroutines back and run the test?

brigaudet commented:
Looks good, thanks for doing this!

mingxuanwupnnl commented on components/eam/src/chemistry/aerosol/wetdep.F90:
```diff
@@ -5,19 +5,21 @@ module wetdep
 private
 
 public :: wetdepa_v2  ! scavenging codes for very soluble aerosols -- CAM5 version
-public :: clddiag     ! calc of cloudy volume and rain mixing ratio
+!public :: clddiag     ! calc of cloudy volume and rain mixing ratio
```
Just wonder why it is commented out.  Not long used in other modules?

tangshuaiqi replied:
That's right. I will just remove this line.

Commit:
Update wetdep.F90

remove declaring clddiag as public subroutine as it is not used in other modules

tangshuaiqi commented:
All test passed. merge into maint
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel
